etc/generate-envoy-config/envoy.jsonnet

```diff
@@ -0,0 +1,196 @@
+                prefix: 'AWS ',
+              },
+            },
+          ],
+        },
+      },
+    ],
+  },
+  'pachd-identity': {
+    internal_port: 1658,
+    external_port: 30658,
+    service: 'pachd-proxy-backend',
+    routes: [
+      {
+        match: {
+          prefix: '/dex',
+        },
+        route: {
+          cluster: 'pachd-identity',
+          idle_timeout: '60s',
+          timeout: '60s',
+        },
+      },
+    ],
+    health_check: {
+      healthy_threshold: 1,
+      http_health_check: {
+        host: 'localhost',
+        path: '/dex/.well-known/openid-configuration',
+      },
+      interval: '30s',
+      timeout: '10s',
+      unhealthy_threshold: 2,
+      no_traffic_interval: '10s',
+      no_traffic_healthy_interval: '10s',
+    },
+  },
+  'pachd-oidc': {
+    internal_port: 1657,
+    external_port: 30657,
+    service: 'pachd-proxy-backend',
+    routes: [
+      {
+        match: {
+          prefix: '/authorization-code/callback',
+        },
+        route: {
+          cluster: 'pachd-oidc',
+          idle_timeout: '60s',
+          timeout: '60s',
+        },
+      },
+    ],
+  },
+  console: {
+    internal_port: 4000,
+    external_port: 4000,
+    service: 'console-proxy-backend',
+    routes: [
+      {
+        match: {
+          prefix: '/',
+        },
+        route: {
+          cluster: 'console',
+          idle_timeout: '600s',
+          timeout: '3600s',
+          upgrade_configs: [
+            {
+              enabled: true,
+              upgrade_type: 'websocket',
+            },
+          ],
+        },
+      },
+    ],
+    health_check: {
+      healthy_threshold: 1,
+      http_health_check: {
+        host: 'localhost',  // This is just the value of the Host: header, not something to connect to.
+        path: '/',
+      },
+      interval: '30s',
+      timeout: '10s',
+      unhealthy_threshold: 2,
+      no_traffic_interval: '10s',
+      no_traffic_healthy_interval: '10s',
+    },
+  },
+  'pachd-metrics': {
+    internal_port: 1656,
+    external_port: 30656,
+    service: 'pachd-proxy-backend',
+    routes: [
+      {
+        match: {
+          prefix: '/',
+        },
+        route: {
+          cluster: 'pachd-metrics',
+          idle_timeout: '60s',
+          timeout: '60s',
+        },
+      },
+    ],
+  },
+};
+
+// This is the generation code.
+local Envoy = import 'envoy.libsonnet';
+Envoy.bootstrap(
+  listeners=[
+    Envoy.httpListener(
+      port=8080,  // Not port 80, because we run as user 101 and not root.
+      name='proxy-http',
+      routes=std.flatMap(function(name) services[name].routes, ['pachd-grpc', 'pachd-s3', 'pachd-identity', 'pachd-oidc', 'console'])
+    ),
+  ] + [
+    local svc = services[name];
+    Envoy.httpListener(
+      port=svc.internal_port,
+      name='direct-' + name,
+      routes=svc.routes,
+    )
+    for name in ['pachd-grpc', 'pachd-s3', 'pachd-identity', 'pachd-oidc', 'console', 'pachd-metrics']
```

Comment:
maybe it would be helpful to clarify which services aren't included here, either by comment or `services - [...]`?
WDYT?

Reply:
I pushed a revision that explains why the multiplexed route table has services listed in a particular order (the priority order in the route table is important).  I made this statement use all the services, added a comment, and made sure that the order doesn't depend on hashtable order.  (It doesn't matter what order the direct listeners show up in, but it will make CI claim that you forgot to regenerate the file if the map key iteration order changes.)